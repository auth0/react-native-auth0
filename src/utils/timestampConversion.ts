export function convertExpiresInToExpiresAt(expiresIn: number): number | null {
    if(expiresIn === null || expiresIn === undefined) {
        return null
    }
    return Math.floor(Date.now() / 1000 + expiresIn)
}