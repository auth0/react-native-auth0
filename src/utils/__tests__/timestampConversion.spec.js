import { convertExpiresInToExpiresAt, convertUnixTimestampToDate } from '../timestampConversion'

describe('timestamp conversion', () => {
    beforeAll(() => {
        jest
          .useFakeTimers()
          .setSystemTime(new Date('2023-01-01'));
      });
    
    describe('test convertExpiresInToExpiresAt', () => {
        it('should successfully convert', () => {
            let expiresIn = 86400
            const result = convertExpiresInToExpiresAt(expiresIn)
            expect(result).toMatchSnapshot();
        });

        it('should handle zero', () => {
            let expiresIn = 0
            const result = convertExpiresInToExpiresAt(expiresIn)
            expect(result).toMatchSnapshot();
        });

        it('should handle null', () => {
            let expiresIn = null
            const result = convertExpiresInToExpiresAt(expiresIn)
            expect(result).toMatchSnapshot();
        });

        it('should handle undefined', () => {
            let expiresIn = undefined
            const result = convertExpiresInToExpiresAt(expiresIn)
            expect(result).toMatchSnapshot();
        });
    })
});
