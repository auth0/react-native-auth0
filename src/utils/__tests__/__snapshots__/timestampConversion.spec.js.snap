// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`timestamp conversion test convertExpiresInToExpiresAt should handle null 1`] = `null`;

exports[`timestamp conversion test convertExpiresInToExpiresAt should handle undefined 1`] = `null`;

exports[`timestamp conversion test convertExpiresInToExpiresAt should handle zero 1`] = `1672531200`;

exports[`timestamp conversion test convertExpiresInToExpiresAt should successfully convert 1`] = `1672617600`;
